lib/host.js

```diff
@@ -251,14 +267,16 @@ HostConnectionPool.prototype._maybeCreatePool = function (callback) {
       return connections.length < self.coreConnectionsLength;
     },
     function iterator(next) {
-      var c = new Connection(self.address, self.protocolVersion, self.options);
-      //Relay the event idleRequestError
-      c.on('idleRequestError', function (err) {
-        //The pool will emit the event
-        self.emit('idleRequestError', err);
-      });
+      var c = self._createConnection();
```

Comment:
core connections are currently created one at a time, we could optimize this by creating them all at once, although I think in the general case the number of core connections will be pretty low.